Documentation/design-docs/hosting-layer-apis.md

```diff
@@ -143,3 +164,10 @@ int corehost_initialize_context(const host_interface_t *init, corehost_context_c
 Initializes the host context. This calculates everything required to start CoreCLR (but does not actually do so).
 * `init` - struct defining how the host context should be initialized. If the host context is already initialized, this function will check if `init` is compatible with the active context.
 * `context_contract` - if initialization is successful, this is populated with the contract for operating on the initialized host context.
+
+``` C
+int corehost_close_context(corehost_context_contract *context_contract)
+```
+
+Closes the host context. Once closed, any operations targeting the closed context will return an error.
```

Comment:
I would not commit to returning sensible errors, unless we want to build a handle table and so on. I was only thinking of casting the handle as a pointer to a struct. So closed handles are effectively freed pointers.
But this is just a nit - we'll get there during implementation.